internal/k8s/crdcontroller.go

```diff
@@ -303,11 +303,31 @@ var refModelMap = map[string]string{
 
 // checkRefOnController checks whether a provided ref on the controller
 func checkRefOnController(key, refKey, refValue string) error {
-	uri := fmt.Sprintf("/api/%s?name=%s&fields=name,type,labels,created_by", refModelMap[refKey], refValue)
-	clients := avicache.SharedAVIClients()
 
 	// assign the last avi client for ref checks
 	aviClientLen := lib.GetshardSize()
+	clients := avicache.SharedAVIClients()
+
+	if lib.GetCloudType() == lib.CLOUD_AWS {
+		if refModelMap[refKey] == "network" {
+			var rest_response interface{}
+			uri := fmt.Sprintf("/api/%s/%s", refModelMap[refKey], refValue)
+			err := lib.AviGet(clients.AviClient[aviClientLen], uri, &rest_response)
+			if err != nil {
+				utils.AviLog.Warnf("key: %s, msg: Get uri %v returned err %v", key, uri, err)
+				return fmt.Errorf("%s \"%s\" not found on controller", refModelMap[refKey], refValue)
+			} else if rest_response != nil {
+				utils.AviLog.Infof("rest_response: %+v", rest_response)
```

Comment:
Can we make it a bit more descriptive in the log? As in, what was the condition in which this log will be printed?

Reply:
updated the logging.

Reply:
PR got merged before I wanted to comment here, the log still isn't clear. Can we combine above two lines and clearly mention what's wrong here?